### DISCUSSION TITLE
Auto reset on DTR

### DISCUSSION THREAD

**Discussion by GGORG0** [earlephilhower/arduino-pico#2402] [Q&A]
Hi, I'd like to make use of the automatic reset on DTR line feature found on Arduinos. 
Essentially, if you connect via serial, upon opening the connection the DTR line will pulse and the ATMega will be reset.
Is there an equivalent feature for the rp2040? Is it possible to implement it on my own, for example by attaching an interrupt to the virtual DTR pin (i know that the rp2040 uses USB-CDC for its serial connection)

If that's not possible, then is there an easy way to reset the rp2040 using the serial connection, without just sending a special packet interpreted by the program over the data lines?

**Comment by earlephilhower** [comment]
It's already there for USB, see: https://github.com/earlephilhower/arduino-pico/blob/bc7adf42c2f4fc6e448bb31c21cef07e658e9289/cores/rp2040/SerialUSB.cpp#L182-L221
